### PR TITLE
DROOLS-2297: Context Editor inline controls

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2297

Video [here](https://youtu.be/FDJQm1oCvz4). This shows the scope of this PR.. different menus on different cells.

The "default" row for a Context has no menus (as it cannot be altered by the User).

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by manstis** [comment]
```
06:52:24 [ERROR] Failed to execute goal io.takari.maven.plugins:takari-lifecycle-plugin:1.12.6:compile (default-compile) on project kie-maven-plugin: 1 error(s) encountered:
06:52:24 [ERROR] /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/droolsjbpm-integration/kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateModelMojo.java:[129:23] no suitable method found for buildAll(CanonicalM[...]::new)
06:52:24 [ERROR]     method org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(java.lang.Class<? extends org.kie.api.builder.KieBuilder.ProjectType>) is not applicable
06:52:24 [ERROR]       (argument mismatch; java.lang.Class is not a functional interface)
06:52:24 [ERROR]     method org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(java.util.function.Predicate<java.lang.String>) is not applicable
06:52:24 [ERROR]       (argument mismatch; invalid constructor reference
06:52:24 [ERROR]           constructor CanonicalModelMavenPluginKieProject in class org.drools.modelcompiler.builder.CanonicalModelMavenPluginKieProject cannot be applied to given types
06:52:24 [ERROR]             required: org.drools.compiler.kie.builder.impl.InternalKieModule,java.lang.ClassLoader
06:52:24 [ERROR]             found: java.lang.String
06:52:24 [ERROR]             reason: actual and formal argument lists differ in length)
```

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/context/AddContextEntryCommand.java [hunk 1]
```diff
@@ -107,10 +115,24 @@ public AddContextEntryCommand(final Context context,
...
         };
     }
+
+    public void updateRowNumbers() {
+        CommandUtils.updateRowNumbers(uiModel,
+                                      IntStream.range(0,
+                                                      uiModel.getRowCount() - 1));
+    }
+
+    public void updateParentInformation() {
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/context/AddContextEntryCommand.java]: I didn't want them public either but ```Mockito``` complained horribly when running *all* the tests together (```mvn test```) about the "spied" command classes not being stubbed correctly although it was quite happy running the command related tests on their own. So, unfortunately, ```public``` was added out of necessity to make the tests run without (Mockito) errors.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/resources/i18n/DMNEditorConstants.java [hunk 2]
```diff
@@ -33,4 +33,28 @@
...
+    public static final String ExpressionEditor_InvocationType = "ExpressionEditor.InvocationType";
+
+    @TranslationKey(defaultValue = "")
+    public static final String ContextEditor_InsertContextEntryAbove = "ContextEditor.InsertContextEntryAbove";
+
+    @TranslationKey(defaultValue = "")
+    public static final String ContextEditor_InsertContextEntryBelow = "ContextEditor.InsertContextEntryBelow";
+
+    @TranslationKey(defaultValue = "")
+    public static final String ContextEditor_DeleteContextEntry = "ContextEditor.DeleteContextEntry";
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/resources/i18n/DMNEditorConstants.java]: Changed the translation to "Delete" (less refactoring - the methods in ```ContextGrid``` are "delete" and - locally - I have a ```DeleteContextEntryCommand``` too... at least it is now consistent!

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**